build/_k-standard-goals.shade

```diff
@@ -83,26 +83,34 @@ default Configuration='${E("Configuration")}'
 
 #native-compile target='compile' if='!IsMono && Directory.Exists(Path.Combine(BASE_DIR, "src"))'
   var programFilesX86 = '${Environment.GetFolderPath(Environment.SpecialFolder.ProgramFilesX86)}'
-  var msbuild = '${Path.Combine(programFilesX86, "MSBuild", "12.0", "Bin", "MSBuild.exe")}'
   var nativeProjects ='${Files.Include(Path.Combine(BASE_DIR, "src", "**", "*.vcxproj"))}'
 
   @{
-      if (nativeProjects.Any())
+    if (nativeProjects.Any())
+    {
+      var msbuildVersions = new[] { "14.0", "12.0"};
+
+      for (var i = 0; i < msbuildVersions.Length; i++)
       {
-        if (!File.Exists(msbuild))
+        var msbuildPath =  Path.Combine(programFilesX86, "MSBuild", msbuildVersions[i], "Bin", "MSBuild.exe");
+        if (File.Exists(msbuildPath))
         {
-            Log.Warn("msbuild version 12 not found. Please ensure you have the VS 2013 C++ SDK installed.");
-            Environment.Exit(1);
+          foreach (var project in nativeProjects)
+          {
+            Exec(msbuildPath, project + " /p:Configuration=" + Configuration + ";Platform=Win32");
+            Exec(msbuildPath, project + " /p:Configuration=" + Configuration + ";Platform=x64");
+          }
+
+          break;
         }
-        else
+
+        if (i == msbuildVersions.Length - 1)
```

Comment:
the loop runs until `i < msbuildVersions.Length`. Because the if is inside loop it will never get to `msbuildVersions.Length`